lamindb/models/record.py

```diff
@@ -843,20 +844,33 @@ def save(self, *args, **kwargs) -> Record:
                 # save unversioned record
                 else:
                     super().save(*args, **kwargs)
-            except IntegrityError as e:
+            except (IntegrityError, ProgrammingError) as e:
                 error_msg = str(e)
                 # two possible error messages for hash duplication
                 # "duplicate key value violates unique constraint"
                 # "UNIQUE constraint failed"
                 if (
-                    "UNIQUE constraint failed" in error_msg
-                    or "duplicate key value violates unique constraint" in error_msg
-                ) and "hash" in error_msg:
+                    isinstance(e, IntegrityError)
+                    and "hash" in error_msg
+                    and (
+                        "UNIQUE constraint failed" in error_msg
+                        or "duplicate key value violates unique constraint" in error_msg
+                    )
+                ):
                     pre_existing_record = self.__class__.get(hash=self.hash)
                     logger.warning(
                         f"returning {self.__class__.__name__.lower()} with same hash: {pre_existing_record}"
                     )
                     init_self_from_db(self, pre_existing_record)
+                elif (
+                    isinstance(e, ProgrammingError)
```

Comment:
Why not using the "PermissionError" instead of "ProgramminError"?

Reply:
InsufficientPrivilege error I mean ☺️

Reply:
Because `django` throws `ProgrammingError` actually, `InsufficientPrivilege` is not the final one, it is an intermediate from `psycopg2`.